indra/llcommon/lua_function.h

```diff
@@ -0,0 +1,253 @@
+/**
+ * @file   lua_function.h
+ * @author Nat Goodspeed
+ * @date   2024-02-05
+ * @brief  Definitions useful for coding a new Luau entry point into C++
+ * 
+ * $LicenseInfo:firstyear=2024&license=viewerlgpl$
+ * Copyright (c) 2024, Linden Research, Inc.
+ * $/LicenseInfo$
+ */
+
+#if ! defined(LL_LUA_FUNCTION_H)
+#define LL_LUA_FUNCTION_H
+
+#include "luau/luacode.h"
+#include "luau/lua.h"
+#include "luau/luaconf.h"
+#include "luau/lualib.h"
+#include "stringize.h"
+#include <exception>                // std::uncaught_exceptions()
+#include <memory>                   // std::shared_ptr
+#include <utility>                  // std::pair
+
+class LuaListener;
+
+namespace lluau
+{
+    // luau defines luaL_error() as void, but we want to use the Lua idiom of
+    // 'return error(...)'. Wrap luaL_error() in an int function.
+#if __clang__
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wformat-security"
```

Comment:
clang isn't all wrong about this. Can a user controlled string enter here or will all of this be sanitized?

Reply:
No, users would directly call [Luau's `error()` function](https://luau-lang.org/library#global-functions). This is for our internal use.

What I would like to know is what magic attribute to associate with the `const char* format` argument to (a) direct the compiler's static validation to _callers_ of this `error()` function, and (b) pass through the statically-validated `format` to the underlying function.

Reply:
If there is no "user" (user in this case could also be resources loaded from XML) supplied char const* (and I understood there is no such thing), then it is indeed strange that clang gets so lippy.

For some warnings  compiler will try track arguments during the function calls and then only warn if they can determine the error. As this function is a template there is a good change clang will do it just like that (this would be visible in the way clang emits the warning).

A plausible explanation could then be attributed to the lua precompiled libs somehow in the call stack, then there can be no verification > error.

For b) then compiling lua inside the viewer source tree could fix it (not a super good solution).

Long story short: I think the pragma is the most pragmatic way then. (Except it will not help with GCC)